src/EventListener/WriteListener.php

```diff
@@ -78,8 +75,7 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
                     $hasOutput = false !== $resourceMetadata->getOperationAttribute($attributes, 'output_class', null, true);
                 }
 
-                $class = \get_class($controllerResult);
-                if ($hasOutput && $attributes['resource_class'] === $class && $class === \get_class($event->getControllerResult())) {
+                if ($hasOutput) {
```

Comment:
how about 
```php
if ($hasOutput && ResourceClassResolver->isResourceClass(\get_class($event->getControllerResult())) {`
```

?

Reply:
We're changing this code

Reply:
@soyuka any comments here after your input/output dto changes?

Reply:
I added a behat test for this particular case so if it doesn't break this looks okay!